airflow/models/dag.py

```diff
@@ -1468,6 +1468,10 @@ def create_dagrun(self,
         :param session: database session
         :type session: sqlalchemy.orm.session.Session
         """
+        # Resolve relationship between task set by XComArgs
+        for task in self.tasks:
+            task.set_xcomargs_dependencies()
```

Comment:
This is going to be too late to affect the Graph view of the webserver isn't it?

(This either needs to go much, much earlier, to dag creation time so it is correctly reflected in the serialized representation, or this is not needed here at all.)

Reply:
That's my main concern. I will add tests for serialization and see what can be done. That was my main concern, that we have to keep this logic in more than one place (dag bag and somewhere around dag creation without dag bag)

Reply:
I removed this code. Everything works fine (dag rendering in web UI, running dags, backfill etc). The only problem is that code like this will not work anymore:
```
if __name__ == '__main__':
    dag.run()
```